python/GafferUI/ColorChooserPlugValueWidget.py

```diff
@@ -89,6 +126,34 @@ def __colorChanged( self, colorChooser, reason ) :
 				for plug in self.getPlugs() :
 					plug.setValue( self.__colorChooser.getColor() )
 
+	def __colorChooserOptionChanged( self, colorChooser, value, key ) :
+
+		if Gaffer.Metadata.value( "colorChooser:inlineOptions", "userDefault" ) is None :
+			sessionOptions = Gaffer.Metadata.value( "colorChooser:inlineOptions", "sessionDefault" )
+			if sessionOptions is None :
+				sessionOptions = IECore.CompoundData()
+				Gaffer.Metadata.registerValue( "colorChooser:inlineOptions", "sessionDefault", sessionOptions )
+
+			sessionOptions.update( { key: value } )
+
+		for p in self.getPlugs() :
+			plugOptions = Gaffer.Metadata.value( p, "colorChooser:inlineOptions" )
+			if plugOptions is None :
+				plugOptions = IECore.CompoundData()
+				Gaffer.Metadata.registerValue( p, "colorChooser:inlineOptions", plugOptions, persistent = False )
+
+			plugOptions.update( { key: value } )
+
+	def __colorChooserOptions( self ) :
+
+		v = sole( Gaffer.Metadata.value( p, "colorChooser:inlineOptions" ) for p in self.getPlugs() )
+		if v is None :
+			v  = Gaffer.Metadata.value( "colorChooser:inlineOptions", "userDefault" )
```

Comment:
`userDefault` has quite a specific meaning already - a default value assigned to plugs on newly-created nodes - and I don't think the semantics match what we're doing here.

I think this is redundant anyway though - we can achieve the same thing using the existing metadata inheritance rules. Registering against `Color3fPlug` and `Color4fPlug` gives us the fallback behaviour, and we can also use node-specific registrations to provide an in-between level of control.

Reply:
I did a big overhaul of how this works for this and the following comment. It does feel a lot more streamlined now. We're only reading from the plug value, which will fall back to `Color3fPlug`, `Color4fPlug` if the plug instance doesn't have an opinion. Done in https://github.com/GafferHQ/gaffer/pull/6003/commits/8bf9543bafcf26dd41e265f7f18376a2d8460a4a